scripts/gen_lsp.lua

```diff
@@ -35,7 +35,10 @@ end
 
 -- Gets the Lua symbol for a given fully-qualified LSP method name.
 local function name(s)
-  return s:gsub('/', '_', 3)
+  if vim.startswith(s, '$') then
+    s = s:gsub('^%$', 'dollar')
+  end
+  return s:gsub('/', '_')
```

Comment:
This can be simplified, startswith() is redundant because `^` in the pattern already checks that.

```suggestion
  -- "$/" prefix is special: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#dollarRequests
  return s:gsub('^%$', 'dollar'):gsub('/', '_')
```